packages/vite/src/node/build.ts

```diff
@@ -406,7 +425,81 @@ async function doBuild(
             ? createMoveToVendorChunkFn(config)
             : undefined,
         ...output
+      }
+    }
+
+    // resolve lib mode outputs
+    const outputs = resolveBuildOutputs(
+      options.rollupOptions?.output,
+      libOptions,
+      config.logger
+    )
+
+    // watch file changes with rollup
+    if (config.build.watch) {
+      config.logger.info(chalk.cyanBright(`\nwatching for file changes...`))
+
+      const output: OutputOptions[] = []
+      if (Array.isArray(outputs)) {
+        for (const resolvedOutput of outputs) {
+          output.push(buildOuputOptions(resolvedOutput))
+        }
+      } else {
+        output.push(buildOuputOptions(outputs))
+      }
+
+      const watcherOptions = config.build.watch
+      const watcher = rollup.watch({
+        ...rollupOptions,
+        output,
+        watch: {
+          ...watcherOptions,
+          chokidar: {
+            ignored: [
+              '**/node_modules/**',
+              '**/.git/**',
+              ...(watcherOptions?.chokidar?.ignored || [])
+            ],
+            ignoreInitial: true,
+            ignorePermissionErrors: true,
+            ...watcherOptions.chokidar
+          }
+        }
+      })
+
+      watcher.on('event', (event) => {
+        if (event.code === 'BUNDLE_START') {
+          config.logger.info(chalk.cyanBright(`\nbuild started...`))
+
+          // clean previous files
+          if (options.write) {
+            emptyDir(outDir)
+            if (fs.existsSync(config.publicDir)) {
+              copyDir(config.publicDir, outDir)
+            }
+          }
+        } else if (event.code === 'BUNDLE_END') {
+          event.result.close()
+          config.logger.info(chalk.cyanBright(`built in ${event.duration}ms.`))
+        } else if (event.code === 'ERROR') {
+          outputBuildError(event.error)
+        }
       })
+
+      // stop watching
+      watcher.close()
```

Comment:
Why is `close` called immediately?

Reply:
Looks like this `close` call is a no-op. Is this line accidental?

See here:
https://github.com/rollup/rollup/blob/dba6f13132a1d7dac507d5056399d8af0eed6375/src/watch/watch-proxy.ts#L16
https://github.com/rollup/rollup/blob/dba6f13132a1d7dac507d5056399d8af0eed6375/src/watch/watch-proxy.ts#L31-L33
https://github.com/rollup/rollup/blob/dba6f13132a1d7dac507d5056399d8af0eed6375/src/watch/watch.ts#L47